pyanaconda/packaging/dnfpayload.py

```diff
@@ -797,23 +795,20 @@ def preInstall(self, packages=None, groups=None):
             self.requiredPackages += packages
         self.requiredGroups = groups
 
-        # Write the langpacks config
-        pyanaconda.iutil.mkdirChain(DNF_PLUGINCONF_DIR)
-        langs = [self.data.lang.lang] + self.data.lang.addsupport
-
-        # Start with the file in /etc, if one exists. Otherwise make an empty config
-        if os.path.exists(_DNF_TARGET_LANGPACK_CONF):
-            shutil.copy2(_DNF_TARGET_LANGPACK_CONF, _DNF_INSTALLER_LANGPACK_CONF)
-        else:
-            with open(_DNF_INSTALLER_LANGPACK_CONF, "w") as f:
-                f.write("[main]\n")
-
-        # langpacks.conf is an INI style config file, read it and
-        # add or change the enabled and langpack_locales entries without
-        # changing anything else.
-        keys=[("langpack_locales", "langpack_locales=" + ", ".join(langs)),
-              ("enabled", "enabled=1")]
-        simple_replace(_DNF_INSTALLER_LANGPACK_CONF, keys)
+        # get all available languages in repos
+        available_langpacks = self._base.sack.query().available() \
+            .filter(name__glob="langpacks-*")
+        alangs = map(lambda p: p.name.split('-', 1)[1], available_langpacks)
```

Comment:
I don't think this would work. ``map()`` returns a ``map`` object that behaves like a generator. If you iterate over it twice, you'll get no results in the second run. You can either call ``list()`` on the ``map()`` result or (preferably) use a list comprehension.